### ISSUE TITLE
Shell js Loading Problem

### ISSUE THREAD

**Issue by mdshohelrana** [leniel/SPACodeCamper#1]
Hi leniel.
I am new in angularjs and breezejs. i have downloaded your SPACodeCamper project and followed your suggestion and try to run. When i run this application shell js can not load all funtion of common js. here i uploaded two image file one is shell js and another is toastr error message. In second image file the debugger point indicate the error line. when this line execute then getLogFn is undefined. I think it would be help you to understand the cause of error. i can mention that this error occurs to IE and Google chrome. Thanks and regards.

![error1](https://cloud.githubusercontent.com/assets/6725033/3049263/aab9d36a-e15a-11e3-8c85-725fd5045cf6.png)
![error2](https://cloud.githubusercontent.com/assets/6725033/3049264/aac16a30-e15a-11e3-852c-0126ce3c3169.png)

**Comment by leniel** [comment]
Hi Shohel,

Oops... incredible! I'm doing now [Building Apps with Angular and Breeze - Part 2](http://pluralsight.com/training/Courses/TableOfContents/build-apps-angular-breeze-part2) and while updating the app and trying to run I got the exact same error you describe.

The problem is hard to spot indeed. I modified the code while doing the [Part 1 course](http://pluralsight.com/training/Courses/TableOfContents/build-apps-angular-breeze) and when passing the dependencies in `function shell(...` I forgot to pass `$q`. See how it was before:

``` JS
var controllerId = 'shell';
    angular.module('app').controller(controllerId,
        ['$rootScope', '$window', '$q', 'common', 'config', shell]);

    function shell($rootScope, $window, common, config)
    {
```

and after:

``` JS
var controllerId = 'shell';
    angular.module('app').controller(controllerId,
        ['$rootScope', '$window', '$q', 'common', 'config', shell]);

    function shell($rootScope, $window, $q, common, config)
    {
```

The problem is that `common` where `logger` resides was being passed `$q`. This way `common` was never passed to `shell()`.

I'll fix it in the next commit.
